#!/usr/bin/env python
PACKAGE = "auto_nav"
 
from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()

gen.add( "global_frame",    str_t,     0,     "global_map_frame", "map")
gen.add( "plan_frequency", int_t,  0,     "planner_frequency",   50,      1,       100)
gen.add( "max_x_speed",    double_t,     0,     "max_x_speed",   0.5,    0.1,       2.0)
gen.add( "max_y_speed",   double_t,    0,     "max_y_speed",      0.5,    0.1,       2.0)
gen.add( "max_yaw_speed", double_t,    0,   "max_yaw_speed ",     0.5,    0,  	5)
gen.add( "set_yaw_speed",  double_t,    0,   "yaw_angular_speed",  1.0,    0,         15)
gen.add( "goal_dist_tolerance", double_t , 0 ,"goal distance tolerance m",0.25, 0.1, 5)
gen.add( "slow_down_tolerance_", double_t , 0 ,"slow down distance tolerance m",0.7, 0.1, 5)
gen.add( "prune_ahead_distance",double_t , 0 ,"prune_ahead_distance m",0.6     , 0.1,  3)
gen.add( "max_angle_diff", double_t,0,    "max_angle_diff du" , 	60,	5,	180)

gen.add( "race_sw",bool_t,0,"true for race,false for debug",True) 

gen.add("p_value",double_t,0,"pid position parameter-p",1.5,0,10)
gen.add("i_value",double_t,0,"pid position parameter-i",1.0,0,10)
gen.add("d_value",double_t,0,"pid position parameter-d",1.0,0,10)

gen.add("vx_accel_limit",double_t,0,"vx_accel_limit",0.05,0,10)
gen.add("vy_accel_limit",double_t,0,"vy_accel_limit",0.05,0,10)


mode_enum = gen.enum([ gen.const("Pure_Pursuit",    int_t, 0, "A small constant"),
                       gen.const("PID_Position",     int_t, 1, "A medium constant"),],
                       "An enum to set track mode")
 
gen.add("track_mode", int_t, 0, "A mode parameter", 1, 0, 1, edit_method=mode_enum)
 
exit(gen.generate(PACKAGE, "auto_nav", "tracks_follow")) 

